addons/pos_online_payment/models/pos_payment_method.py

```diff
@@ -0,0 +1,122 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+
+class PosPaymentMethod(models.Model):
+    _inherit = "pos.payment.method"
+
+    is_online_payment = fields.Boolean(string="Online Payment", help="Use this payment method for online payments (payments made on a web page with online payment providers)", default=False)
+    online_payment_provider_ids = fields.Many2many('payment.provider', string="Allowed Providers", domain="[('is_published', '=', True), ('state', 'in', ['enabled', 'test'])]")
+    has_an_online_payment_provider = fields.Boolean(compute='_compute_has_an_online_payment_provider', readonly=True)
+    type = fields.Selection(selection_add=[('online', 'Online')])
+
+    @api.depends('is_online_payment', 'journal_id', 'split_transactions')
+    def _compute_type(self):
+        opm = self.filtered(lambda pm: pm.is_online_payment)
+        if opm:
+            opm.update({'type': 'online'})
+
+        super(PosPaymentMethod, self - opm)._compute_type()
+
+    def _get_online_payment_providers(self, pos_config_id=False, error_if_invalid=True):
+        self.ensure_one()
+        providers_sudo = self.sudo().online_payment_provider_ids
+        if len(providers_sudo) <= 0: # Empty = all published providers
+            providers_sudo = self.sudo().env['payment.provider'].search([('is_published', '=', True), ('state', 'in', ['enabled', 'test'])])
+
+        if not pos_config_id:
+            return providers_sudo
+
+        config_currency = self.sudo().env['pos.config'].browse(pos_config_id).currency_id
+        valid_providers = providers_sudo.filtered(lambda p: not p.journal_id.currency_id or p.journal_id.currency_id == config_currency)
+        if error_if_invalid and len(providers_sudo) != len(valid_providers):
+            raise ValidationError(_("All payment providers configured for an online payment method must use the same currency as the Sales Journal, or the company currency if that is not set, of the POS config."))
+        return valid_providers
+
+    @api.depends('is_online_payment', 'online_payment_provider_ids')
+    def _compute_has_an_online_payment_provider(self):
+        for pm in self:
+            if pm.is_online_payment:
+                pm.has_an_online_payment_provider = len(pm._get_online_payment_providers()) > 0
+            else:
+                pm.has_an_online_payment_provider = False
+
+    def _is_write_forbidden(self, fields):
+        whitelisted_fields = set(('online_payment_provider_ids',))
+        return super(PosPaymentMethod, self)._is_write_forbidden(fields - whitelisted_fields)
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            if vals.get('is_online_payment', False):
```

Comment:
Yes, but writting it explicitly shows what I consider: the field "is_online_payment" has a default value of False. Maybe tomorrow that will be changed, and the person who changes it will directly see it.
So yes it's not necessary, but I avoid a conversion of None to False, and I highlight what I consider.

Reply:
> Which matters how?
If the person who changes the default value of is_online_payment and forget to change it in this part of the code, anyone after can spot the issue, understand that it was False before and fix the issue. Otherwise there is the doubt of knowing if the person who wrote "vals.get('is_online_payment')" thought about the default value of the field.

> You're not actually avoiding anything useful. Conditions on None are idiomatic, it's special-cased in all sorts of locations in the interpreter. You're saving a C-level branch, at the cost of adding an instruction (a much larger branching), an array lookup, and increased refcount traffic.
I perfer to don't waste my time on considering performances of such little things. The conversion is mental, the code is meant to be read.